### ISSUE TITLE
Failure message with 'TRUE'

### ISSUE THREAD

**Issue by kelshmo** [Sage-Bionetworks/dccvalidator#284]
A contributor is reporting a failure message in the case they enter 'true' as an enumerated string value (there are keys in the clinical data where true,false,unknown are enumerated values). In this case, it looks like one of our `dccvalidator` import functions is reading the key as a logical and changing it to TRUE.
<img width="477" alt="Screen Shot 2020-02-13 at 10 03 25 AM" src="https://user-images.githubusercontent.com/40647130/74464267-15e28500-4e48-11ea-86b8-93891b24ae66.png">

Possible solutions: 
- debug why the validator converts values to `TRUE` and prevent this <- don't think this is at all feasible, this would require some super intelligence to pre-define column types and we want to be able to slot in any schema
- enumerate all alternative capitalization as acceptable values in the annotations table/schema

**Comment by karawoo** [comment]
> don't think this is at all feasible, this would require some super intelligence to pre-define column types and we want to be able to slot in any schema

Correct, this would be very hard. We can override column guesses when reading in the data, but only if we know in advance what the columns will be. This would create a huge amount of work to choose the right specification for each possible template. I wish there were a way to just override all logicals and make them all character, but I don't think there is 😕 

Another option is to read everything as character with `col_types = cols(.default = "c")`. Then we would have to go back and convert all numeric characters to numeric after the data is read in.

**Comment by kelshmo** [comment]
> Another option is to read everything as character with `col_types = cols(.default = "c")`. Then we would have to go back and convert all numeric characters to numeric after the data is read in.

Would this have any unintended impact on validating logical values? Column types make my head spin... 
I read though our [coercibility logic](https://github.com/Sage-Bionetworks/dccvalidator/blob/b46827c4f3f151d3b368e7b0181f5e5f3bfbe463/R/check-annotation-values.R#L300-L302) and if I am conceptualizing this correctly, it sounds like importing logicals/booleans as character should work OK.

**Comment by Aryllen** [comment]
> Another option is to read everything as character with `col_types = cols(.default = "c")`. Then we would have to go back and convert all numeric characters to numeric after the data is read in.

Converting all numeric characters into numeric sounds icky, but this might be the simplest way.

> Would this have any unintended impact on validating logical values?

I don't believe so since the `can_coerce` logic uses a set of acceptable strings for validation.

**Comment by karawoo** [comment]
Yeah, I think it would be ok for validating logical values. We'd definitely want to have tests for this case though.

**Comment by kelshmo** [comment]
There are currently 5 key-value pairs that fall into this use case - I'm tempted to consider enumerating alternative capitalization as the best use of time/resources.

**Comment by karawoo** [comment]
That's fine by me if it works for you @kelshmo. I may leave this issue open but mark it as low priority in that case.

**Comment by kelshmo** [comment]
We implemented a temporary workaround by enumerating values explicitly.
